### PR TITLE
Remove assert.h include from DxilPipelineStateValidation.h

### LINKED ISSUE #6922: Include <assert.h> in DxilPipelineStateValidation.h caused build failures
**Description**
Some projects outside DXC include DxilPipelineStateValidation.h. Their own custom assert might be overwritten by standard assert if  <assert.h> is included in DxilPipelineStateValidation.h.  

**Actual Behavior**
The compilation will report "too many arguments for function-like macro invocation" if the custom assert is defined has multiple arguments.

**Environment**
- DXC version:  dxcompiler.dll: 1.8 - 1.8.0.14576 (main, 381750e25); dxil.dll: 1.8(1.8.0.14576)
- Host Operating System : Windows

### PR DESCRIPTION
**Pull request by lizhengxing**
DxilPipelineStateValidation.h is included from multiple environments. Don't include assert.h in it, otherwise it might overwrite the custom assert in the code that includes DxilPipelineStateValidation.h.

Fixes #6922

### CODE REVIEW

**Review by damyanp — COMMENTED** [review]
It's unfortunate that we have to do this, but I believe that this is the pragmatic change to make at this point.

Can you file an issue please to cover cleaning this up properly?

File: lib/DxilContainer/DxilPipelineStateValidation.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #include "llvm/IR/DataLayout.h"
 #include "llvm/IR/Module.h"
 #include "llvm/Support/raw_ostream.h"
+#include <assert.h> // Needed for DxilPipelineStateValidation.h
```
Comment (tex3d) [inline lib/DxilContainer/DxilPipelineStateValidation.cpp:23]: clang-format autopublish moved this assert here, but it needs to be included before `DxilPipelineStateValidation.h`.  So I believe it needs to be separated by a blank line, right?

File: lib/DxilContainer/DxilRDATBuilder.cpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
 #include "dxc/DxilContainer/DxilPipelineStateValidation.h"
 #include "dxc/Support/FileIOHelper.h"
 #include "dxc/Support/Global.h"
+#include <assert.h> // Needed for DxilPipelineStateValidation.h
```
Comment (tex3d) [inline lib/DxilContainer/DxilRDATBuilder.cpp:5]: clang-format autopublish moved this assert here, but it needs to be included before `DxilPipelineStateValidation.h`.  So I believe it needs to be separated by a blank line, right?

File: lib/DxilContainer/DxilPipelineStateValidation.cpp (hunk 1 above)
Comment (lizhengxing) [reply lib/DxilContainer/DxilPipelineStateValidation.cpp]: @tex3d  Yes, you're right. Added a blank line to fix it.

File: lib/DxilContainer/DxilRDATBuilder.cpp (hunk 2 above)
Comment (lizhengxing) [reply lib/DxilContainer/DxilRDATBuilder.cpp]: @tex3d Yes, you're right. Added a blank line to fix it.

File: include/dxc/DxilContainer/DxilPipelineStateValidation.h [hunk 3]
```diff
@@ -12,20 +12,19 @@
 #ifndef __DXIL_PIPELINE_STATE_VALIDATION__H__
 #define __DXIL_PIPELINE_STATE_VALIDATION__H__
 
+// Don't include assert.h here.
+// Since this header is included from multiple environments,
+// it is necessary to define assert before this header is included.
+// #include <assert.h>
```
Comment (python3kgae) [inline include/dxc/DxilContainer/DxilPipelineStateValidation.h:18]: Shall we remove `// #include <assert.h>`?

**Comment by lizhengxing** [comment]
> It's unfortunate that we have to do this, but I believe that this is the pragmatic change to make at this point.
> 
> Can you file an issue please to cover cleaning this up properly?

Sure, I've filed an issue here #6922

File: include/dxc/DxilContainer/DxilPipelineStateValidation.h (hunk 3 above)
Comment (lizhengxing) [reply include/dxc/DxilContainer/DxilPipelineStateValidation.h:18]: I think it's fine. It's the original comment and use `// #include <assert.h>` here probably can catch more attention.

**Comment by lizhengxing** [comment]
> It's unfortunate that we have to do this, but I believe that this is the pragmatic change to make at this point.
> 
> Can you file an issue please to cover cleaning this up properly?

@damyanp Sure, I've filed an issue here https://github.com/microsoft/DirectXShaderCompiler/issues/6922

**Review by python3kgae — APPROVED** [review]

**Review by damyanp — CHANGES_REQUESTED** [review]
Can you investigate whether we could do something like:

```
#ifndef assert
#include <assert.h>
#endif
```

So that we can remove all the modifications to the other files, but still have this header worked when used in the other environments please?

**Comment by lizhengxing** [comment]
Close this PR since we will use this fix https://github.com/microsoft/DirectXShaderCompiler/pull/6925 .

**Pull request closed without merging**